src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/HeaderTimeChecksPosRule.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Consensus.Rules;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.Consensus.Rules.CommonRules
+{
+    /// <summary>
+    /// Checks if <see cref="PosBlock"/> has a valid time stamp.
```

Comment:
Not exactly, it is just part of the whole checking, so I would say something like if the timestamp is strictly greater than previous block timestamp